### PR TITLE
Document /source/{project_name}/{package_name}/{file_name} endpoints with OpenAPI

### PR DESCRIPTION
**Pull request by ncounter**
### For reviewers

Access to the documented endpoint in the review-app through [this link](https://obs-reviewlab.opensuse.org/eduardoj-apidocs_newsource_package_get/apidocs-new/index.html#/Sources%20-%20Files/get_source__project_name___package_name_).

### Test

**GET**
- run `curl -u foo:bar -H 'Content-Type: application/xml; charset=utf-8' -X GET 'http://localhost:3000/source/home:Admin/hello_world/hello_world.changes'` to retrieve the content of the `hello_world.changes` file
- check if the content of the file matches the webUI one

**DELETE**
- run `curl -u foo:bar -H 'Content-Type: application/xml; charset=utf-8' -X DELETE 'http://localhost:3000/source/home:Admin/hello_world/hello_world.changes'` to delete the `hello_world.changes` file
- check if the file got deleted (either in the webUI, or by retrieving the package's file list with `curl -u foo:bar -H 'Content-Type: application/xml; charset=utf-8' -X DELETE 'http://localhost:3000/source/home:Admin/hello_world'`

**PUT**
- run `curl -u foo:bar -H 'Content-Type: application/xml; charset=utf-8' -d 'Some text for the content' -X PUT 'http://localhost:3000/source/home:Admin/hello_world/new_file'` in order to create a new file
- check if the file got created (either in the webUI, or by retrieving the package's file list with `curl -u foo:bar -H 'Content-Type: application/xml; charset=utf-8' -X GET 'http://localhost:3000/source/home:Admin/hello_world'`)

### CODE REVIEW

**Comment by ncounter** [comment]
[Sources - Files (M)](https://trello.com/c/ONhpe6yD/1988-sources-files-m)

File: src/api/public/apidocs-new/paths/source_project_name_package_name_file_name.yaml [hunk 1]
```diff
@@ -58,3 +58,46 @@ get:
       $ref: '../components/responses/unknown_project_or_package_or_file.yaml'
   tags:
     - Sources - Files
+
+delete:
+  summary: Deletes a specified source file of a package.
+  description: Deletes a specified source file of a package of a project.
+  security:
+    - basic_authentication: []
+  parameters:
```
Comment (eduardoj) [inline src/api/public/apidocs-new/paths/source_project_name_package_name_file_name.yaml:67]: I miss the `meta` parameter.

File: src/api/public/apidocs-new/paths/source_project_name_package_name_file_name.yaml [hunk 2]
```diff
@@ -101,3 +101,59 @@ delete:
       $ref: '../components/responses/unknown_project_or_package_or_file.yaml'
   tags:
     - Sources - Files
+
+put:
+  summary: Write a specified source file into a package.
+  description: Write a specified source file of a package of a project.
+  security:
+    - basic_authentication: []
+  parameters:
```
Comment (eduardoj) [inline src/api/public/apidocs-new/paths/source_project_name_package_name_file_name.yaml:110]: I miss the `keeplink` and `meta` parameters.

File: src/api/public/apidocs-new/components/responses/unknown_parameter.yaml [hunk 3]
```diff
@@ -0,0 +1,10 @@
+description: Unknown parameter
+content:
+  application/xml; charset=utf-8:
+    schema:
+      $ref: '../schemas/api_response.yaml'
+    examples:
+      Unknown parameter:
+        value:
+          code: unknown_parameter
+          summary: "unknown parameter 'limit'"
```
Comment (eduardoj) [inline src/api/public/apidocs-new/components/responses/unknown_parameter.yaml:10]: I would not create this file if this component is going to be used only once, or there isn't any plan to reuse it.

File: src/api/public/apidocs-new/paths/source_project_name_package_name_file_name.yaml [hunk 4]
```diff
@@ -0,0 +1,60 @@
+get:
+  summary: Read a specified source file
+  description: Read a specified source file and return the content
+  security:
+    - basic_authentication: []
```
Comment (saraycp) [inline src/api/public/apidocs-new/paths/source_project_name_package_name_file_name.yaml:5]: I'd say this is not needed for the `get` one. Everybody can see the content of a file, AFAIK.

**Review by saraycp — APPROVED** [review]

**Pull request merged**